lil-lab/recnet
# [recnet-web][v1.10.2] Improve error UI

## Description


Previously, when API is down, the logged-in user will be directed to `onboard` page since we can't get user data from api, which is a weird behavior. We should have a error UI when API is down to let user know what's going on and prevent unexpected action at the same time.

## Related Issue





- #215 

## Notes



## Screenshots

![Screenshot 2024-09-06 at 11 51 18 PM](https://github.com/user-attachments/assets/86a0eacb-90b2-4dd2-8974-3d357c76736f)

When API is down, all pages will be unaccessible (show the error UI above) except landing page, `/help` page, and `/about` page.

## Test

Since it's not testable at staging site, you can test this locally instead. Here are steps for manual test:

1. Run the FE service `nx dev recnet` and don't run the api service to simulate server down time
2. Go to page other than landing page, `/help` page, and `/about` page. 
3. Should see the error UI like above.

## TODO
